portal/celery_worker.py

```diff
@@ -26,26 +26,31 @@ def setup_periodic_tasks(sender, **kwargs):
     """Configure scheduled jobs in Celery"""
     from .tasks import logger
     logger.info("Starting ScheduledJob load...")
-    # create test task if non-existent
-    if not ScheduledJob.query.filter_by(name="__test_celery__").first():
-        test_job = ScheduledJob(name="__test_celery__", task="test",
-                                schedule="0 * * * *", active=True)
-        db.session.add(test_job)
-        db.session.commit()
-        test_job = db.session.merge(test_job)
-
-    # add all tasks to Celery
-    logger.info("ScheduledJobs found: {}".format(ScheduledJob.query.count()))
-    for job in ScheduledJob.query.all():
-        task = getattr(tasks, job.task, None)
-        if task:
-            logger.info("Adding task (id=`{}`, task=`{}` "
-                        "to CeleryBeat".format(job.id, job.task))
-            args_in = job.args.split(',') if job.args else []
-            kwargs_in = job.kwargs or {}
-            kwargs_in['job_id'] = job.id
-            try:
-                sender.add_periodic_task(job.crontab_schedule(),
-                                         task.s(*args_in, **kwargs_in))
-            except Exception as exc:
-                logger.error(exc)
+
+    # As this fires from a celery worker, obtain a scoped session to free
+    # the resource, which otherwise prevents test runs from purging the db
+    with db.scoped_session as session:
+        # create test task if non-existent
+        if not session.query(ScheduledJob).filter_by(
+                name="__test_celery__").first():
+            test_job = ScheduledJob(name="__test_celery__", task="test",
+                                    schedule="0 * * * *", active=True)
+            session.add(test_job)
+            session.commit()
+
+        # add all tasks to Celery
+        logger.info("ScheduledJobs found: {}".format(
+            ScheduledJob.query.count()))
+        for job in session.query(ScheduledJob).all():
+            task = getattr(tasks, job.task, None)
+            if task:
```

Comment:
Could be inverted (`if not task; continue`) into a guard clause for less nesting